core/src/com/unciv/models/stats/Stats.kt

```diff
@@ -112,15 +112,26 @@ open class Stats(
         faith * number
     )
 
-    /** Multiplies each value of this instance by [number] in place */
-    fun timesInPlace(number: Float) {
-        production *= number
-        food *= number
-        gold *= number
-        science *= number
-        culture *= number
-        happiness *= number
-        faith *= number
+    /** Multiplies each value of this instance by [number] in place. If [stat] is supplied, only multiples that stat. */
+    fun timesInPlace(number: Float, stat: Stat? = null) {
+        when (stat) {
+            Stat.Production -> production *= number
+            Stat.Food       -> food *= number
+            Stat.Gold       -> gold *= number
+            Stat.Science    -> science *= number
+            Stat.Culture    -> culture *= number
+            Stat.Happiness  -> happiness *= number
+            Stat.Faith      -> faith *= number
+            else -> {
+                production *= number
+                food *= number
+                gold *= number
+                science *= number
+                culture *= number
+                happiness *= number
+                faith *= number
+            }
+        }
```

Comment:
When doing it for a just single stat, can't you just do `Stat.Production *= number`?

Reply:
:scream: Polished performance lost to a conditional. `fun timesInPlace(stat: Stat? = null, number: Float) = if (stat == null) timesInPlace(number) else this[stat] *= number` - at least most existing calls won't hit this overload. Untested though.

Reply:
Moot anyway as Stat.valueOf(unique.params[1]) will crash instead of returning a null